x-pack/plugins/licensing/common/license.test.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { License } from './license';
+import { LICENSE_CHECK_STATE, ElasticsearchError } from '../server/types';
+import { licenseMock } from './license.mock';
+
+describe('License', () => {
+  const basicLicense = licenseMock.create();
+  const basicExpiredLicense = licenseMock.create({ license: { status: 'expired' } });
+  const goldLicense = licenseMock.create({ license: { type: 'gold' } });
+
+  const error = new Error('unavailable');
+  const errorLicense = new License({ error, signature: '' });
+  const errorNotInstalled: ElasticsearchError = new Error('x-pack not installed');
+  errorNotInstalled.status = 400;
+  const errorXpackNotInstalledLicense = new License({ error: errorNotInstalled, signature: '' });
+  const unavailableLicense = new License({ signature: '' });
+
+  it('uid', () => {
+    expect(basicLicense.uid).toBe('uid-000000001234');
+    expect(errorLicense.uid).toBeUndefined();
+    expect(unavailableLicense.uid).toBeUndefined();
+  });
+
+  it('status', () => {
+    expect(basicLicense.status).toBe('active');
+    expect(errorLicense.status).toBeUndefined();
+    expect(unavailableLicense.status).toBeUndefined();
+  });
+
+  it('expiryDateInMillis', () => {
+    expect(basicLicense.expiryDateInMillis).toBe(5000);
+    expect(errorLicense.expiryDateInMillis).toBeUndefined();
+    expect(unavailableLicense.expiryDateInMillis).toBeUndefined();
+  });
+
+  it('type', () => {
+    expect(basicLicense.type).toBe('basic');
+    expect(goldLicense.type).toBe('gold');
+    expect(errorLicense.type).toBeUndefined();
+    expect(unavailableLicense.type).toBeUndefined();
+  });
```

Comment:
These seem to just be testing the fixtures data rather than any logic in the License class itself.

Reply:
I think we should test this condition https://github.com/elastic/kibana/pull/49345/files/10d1d20ffb6c14d746735cefa66456d10d7fe036#diff-cbd43e825571c43c16799ea77eef2504R57-R61